### PR TITLE
HDDS-6946. Bump Hadoop to 3.3.4

### PR DESCRIPTION
**Pull request by adoroszlai**
## What changes were proposed in this pull request?

 * Bump Hadoop 3 dependency to 3.3.4.
 * Exclude its new `reload4j` dependency (see [HADOOP-18088](https://issues.apache.org/jira/browse/HADOOP-18088)).
 * Explicitly add `log4j` dependency in `hdds-server-framework`.

https://issues.apache.org/jira/browse/HDDS-6946

## How was this patch tested?

Regular CI:
https://github.com/adoroszlai/hadoop-ozone/actions/runs/2866410211

### CODE REVIEW

File: hadoop-hdds/hadoop-dependency-server/pom.xml [hunk 1]
```diff
@@ -71,6 +71,22 @@ https://maven.apache.org/xsd/maven-4.0.0.xsd">
...
+          <groupId>log4j</groupId>
+          <artifactId>log4j</artifactId>
+        </exclusion>
+        <exclusion>
+          <groupId>ch.qos.reload4j</groupId>
+          <artifactId>reload4j</artifactId>
+        </exclusion>
+        <exclusion>
+          <groupId>org.slf4j</groupId>
+          <artifactId>slf4j-log4j12</artifactId>
```
Comment (jojochuang) [inline hadoop-hdds/hadoop-dependency-server/pom.xml:84]: i haven't checked but was it necessary to exclude slf4j-log4j12 and log4j?

Comment (adoroszlai) [reply hadoop-hdds/hadoop-dependency-server/pom.xml:84]: I don't think they are strictly necessary.  They used to be excluded in client dependencies, but not in server dependencies.  I added them here for consistency.

**Review by ayushtkn — COMMENTED** [review]
Please hold it, unless very necessary. I might need to add some ozone code dependencies to hive as part of [HIVE-26360](https://issues.apache.org/jira/browse/HIVE-26360).

And [HADOOP-18033](https://issues.apache.org/jira/browse/HADOOP-18033) doesn't let neither Tez nor Hive move above 3.3.1,
It breaks Spark as well post 3.3.2, there is an attempt to fix spark via [HADOOP-18303](https://issues.apache.org/jira/browse/HADOOP-18303) in 3.3.4, which also doesn't seems to get concluded any time soon

So, this upgrade might create trouble for downstream projects

**Comment by adoroszlai** [comment]
> So, this upgrade might create trouble for downstream projects

Thanks @ayushtkn for the info.  I don't think this is urgent, so marked as draft.

**Review by ayushtkn — APPROVED** [review]
+1, if the build comes green.
Was waiting to see if the PR at TEZ-4420 comes green & unfortunately I forgot about this post that. :-(

**Comment by adoroszlai** [comment]
@jojochuang @smengcl would you like to take a look?

**Review by smengcl — APPROVED** [review]
+1. Thanks @adoroszlai

**Comment by adoroszlai** [comment]
Thanks @ayushtkn, @jojochuang, @smengcl for the review.

**Pull request merged**